presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -281,6 +283,38 @@ public static double round(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(S
         return Math.floor(num * factor + 0.5) / factor;
     }
 
+    @Description("cut to nearest integer")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long cut(@SqlType(StandardTypes.BIGINT) long num)
+    {
+        return cut(num, 0);
+    }
+
+    @Description("cut to nearest integer")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long cut(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.BIGINT) long decimals)
+    {
+        return num;
+    }
+
+    @Description("cut to nearest integer")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double cut(@SqlType(StandardTypes.DOUBLE) double num)
+    {
+        return cut(num, 0);
+    }
+
+    @Description("cut to given number of decimal places")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double cut(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(StandardTypes.BIGINT) long decimals)
+    {
+        return new BigDecimal(num).setScale((int) decimals, RoundingMode.DOWN).doubleValue();
```

Comment:
For rounding mode, I think although mathematically `DOWN` and `HALF_EVEN` (or other `HALF_*`) suffer similar problem in our use case. I always assumed using `DOWN` is more likely to result in problems in real life. Let me know if I'm wrong.